src/main/java/dk/cngroup/trainings/spring/springassignment/controller/web/AnimalWebController.java

```diff
@@ -1,53 +1,150 @@
 package dk.cngroup.trainings.spring.springassignment.controller.web;
 
-import dk.cngroup.trainings.spring.springassignment.exception.InvalidAnimalException;
+import dk.cngroup.trainings.spring.springassignment.exception.*;
 import dk.cngroup.trainings.spring.springassignment.model.Animal;
+import dk.cngroup.trainings.spring.springassignment.model.CareTaker;
 import dk.cngroup.trainings.spring.springassignment.service.AnimalService;
 import org.springframework.stereotype.Controller;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.ModelAttribute;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
 import org.springframework.web.servlet.ModelAndView;
 
 import javax.validation.Valid;
+import java.util.List;
 
 @Controller
-@RequestMapping("/web/animals")
+@RequestMapping("/web/animal")
 public class AnimalWebController {
 
-	private AnimalService animalService;
+	private final AnimalService animalService;
+	private final String VIEW_PATH = "views/animal";
 
 	public AnimalWebController(AnimalService animalService) {
 		this.animalService = animalService;
 	}
 
-	@GetMapping("")
-	public ModelAndView addAnimalView() {
-		ModelAndView modelAndView = new ModelAndView("addanimal");
-		modelAndView.addObject("animals", animalService.getAnimals());
-		modelAndView.addObject("animal", new Animal());
+	@GetMapping("/list")
+	public ModelAndView list() {
+		List<Animal> animals = animalService.getAnimals();
+		ModelAndView modelAndView = new ModelAndView(VIEW_PATH + "/animal_list");
+		modelAndView.addObject("animals", animals);
 		return modelAndView;
 	}
 
-	@PostMapping("")
-	public ModelAndView addAnimal(@ModelAttribute @Valid Animal animal) {
-		ModelAndView modelAndView = new ModelAndView();
-		try {
+	@GetMapping("/add")
+	public ModelAndView add() {
+		ModelAndView modelAndView = new ModelAndView(VIEW_PATH + "/animal_form");
+		Animal animal = new Animal();
+		animal.setId(0L);
```

Comment:
why is this here? it probably should not be here (maybe move it to Animal construtor?

Reply:
Yes I will.